packages/core/src/lib/phrase.ts

```diff
@@ -0,0 +1,46 @@
+import { LanguageKey } from '@logto/core-kit';
+import resource from '@logto/phrases-ui';
+import { ResourceLanguage } from '@logto/phrases-ui/lib/types';
+import deepmerge from 'deepmerge';
+
+import { findAllCustomLanguageKeys, findCustomPhraseByLanguageKey } from '@/queries/custom-phrase';
+
+const isBuiltInLanguage = (key: string): key is LanguageKey => Object.keys(resource).includes(key);
+
+const getCustomPhrase = async (key: string | undefined, customLanguages: string[]) =>
+  key && customLanguages.includes(key) ? findCustomPhraseByLanguageKey(key) : {};
+
+const getBuiltInPhrase = (key: string | undefined) =>
+  key && isBuiltInLanguage(key) ? { ...resource[key], languageKey: key } : {};
+
+const has = (object: Record<string, unknown>) => Object.keys(object).length > 0;
```

Comment:
nit:
`has` seems unclear? maybe using `isNotEmpty()` or something else?
And as this is only used in `getPhrase()`, we may simply defined this method at the beginning of `getPhrase()`?

Reply:
refactored and this function is unneeded.